crates/types/src/transaction/context.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{Layer, State, TransactionId};
+
+/// Holds the `Transaction`'s Context.
+///
+/// Once created it should NOT be modified (immutable).
+/// It contains properties related to the context in which a Transaction will execute.
+/// Additionally, it will encapsulate properties inferred from the binary Transaction sent over-the-wire.
```

Comment:
Good summary. I suggest once change though: Rather than saying that you shouldn't modify `Context`, let's say that you can't. Period. Something like "An immutable container for the `Transactions`'s Context. ... This type does not allow any modifications as that would ... (insert bad stuff here, why we can't allow users to do it)".

As of now, it seems that you *can* modify it even though you shouldn't, while in reality there's no methods that let you do that.

Does this make sense?

Reply:
Yes, also `Context` is written as an immutable type.